core/src/main/scala/org/apache/spark/storage/PushBasedFetchHelper.scala

```diff
@@ -138,10 +140,19 @@ private class PushBasedFetchHelper(
         ((shuffleBlockId.shuffleId, shuffleBlockId.reduceId), size)
     }.toMap
     val address = req.address
+    var remainingBlockCount = sizeMap.size
+    val requestStartTime = clock.nanoTime()
+
     val mergedBlocksMetaListener = new MergedBlocksMetaListener {
       override def onSuccess(shuffleId: Int, reduceId: Int, meta: MergedBlockMeta): Unit = {
         logInfo(s"Received the meta of push-merged block for ($shuffleId, $reduceId)  " +
           s"from ${req.address.host}:${req.address.port}")
+        remainingBlockCount -= 1
+        if (remainingBlockCount == 0) {
+          iterator.logFetchIfSlow(
+            TimeUnit.NANOSECONDS.toMillis(clock.nanoTime() - requestStartTime),
+            sizeMap.values.sum, sizeMap.size, address)
+        }
```

Comment:
For the FetchShuffleBlocks, I think we check `remainingBlock==0` to know when the network request (containing multiple blocks) is done.  @xkrogen  please correct me if I am wrong. 

Here we don't need to because each network req is just requesting the meta of 1 merged block. However, the size of the meta information is not know to the client.  The size here is not the size of the meta information which is being fetched from the server but the size of the merged block. So, I think we can't use that for figuring out if the fetch of this meta information is slow. Hmmm.. Then I think maybe we should not log it for the meta request. Sorry, should have thought about it earlier.

Reply:
> I think we check `remainingBlock==0` to know when the network request (containing multiple blocks) is done.

Yes, that's right.

I also agree that we shouldn't be checking for slowness in the metadata request. @otterc, it seems all of the methods in `PushBasedFetchHelper` for fetching merged blocks are specifically for local ones, is there somewhere we fetch remote merged blocks that we should be logging slowness?

Reply:
The fetch of remote merged block data happens in the `ShuffleBlockFetchIterator.sendRequest`. That either sends `FetchShuffleBlocks` or `FetchShuffleBlockChunks`. So, logging slowness for those are already covered.

Reply:
Oh perfect!

Reply:
Thanks @otterc for digging deeper in this. I must admit that I don't have a good understanding of this codebase yet. Removed the handling for `PushBasedFetchHelper`.